skill-tree/src/test/java/com/RDS/skilltree/EndorsementsIntegrationTests.java

```diff
@@ -290,4 +290,96 @@ public void testAPIReturn404_OnEndorsementNotFound() {
+    }
+
+    @Test
+    @DisplayName("Return 200 with the endorsements of a particular skill given skillID")
+    public void testAPIReturn200_OnEndorsementSearchBySkillID() {
+        String skillIDString = "7a6b8876-44e3-4b18-8579-79e9d4a5f0c9";
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString)
+                .then()
+                    .statusCode(200)
+                    .contentType("application/json")
+                    .body("content", everyItem(hasKey("skill_id")))
+                    .body("content.skill_id", everyItem(equalTo(skillIDString)));
+    }
+
+    @Test
+    @DisplayName("Return 200 with the endorsements matching the given userID and skillID")
+    public void testAPIReturn200_OnEndorsementSearchByBothUserIDAndSkillID() {
+        String userIDString = "73e0b7c4-d128-4e53-9501-0e7f4ff5a261";
+        String skillIDString = "7a6b8876-44e3-4b18-8579-79e9d4a5f0c9";
+
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString + "&userID=" + userIDString)
+                .then()
+                    .statusCode(200)
+                    .contentType("application/json")
+                    .body("content", everyItem(hasKey("skill_id")))
+                    .body("content.skill_id", everyItem(equalTo(skillIDString)))
+                    .body("content", everyItem(hasKey("user_id")))
+                    .body("content.user_id", everyItem(equalTo(userIDString)));
+    }
+
+    @Test
+    @DisplayName("Return 204 when there are no endorsements present for the given userID")
+    public void testAPIReturn204_OnEndorsementSearchWithValidUserID() {
+        String userIDString = UUID.randomUUID().toString();
+
+            given()
+                .when()
+                    .get("/v1/endorsements?userID=" + userIDString)
+                .then()
+                    .statusCode(204);
+    }
+
+    @Test
+    @DisplayName("Return 204 when there are no endorsements present for the given skillID")
+    public void testAPIReturn204_OnEndorsementSearchWithValidSkillID() {
+        String skillIDString = UUID.randomUUID().toString();
+
+            given()
+                .when()
+                    .get("/v1/endorsements?skillID=" + skillIDString)
+                .then()
+                    .statusCode(204);
+    }
+
+    @Test
+    @DisplayName("Return 400 given an invalid userID")
+    public void testAPIReturn400_OnEndorsementSearchWithInvalidUserID() {
+        String userIDString = "invalid-user-id";
+            given()
+                .when()
+                    .get("/v1/endorsements?userID=" + userIDString)
+                .then()
+                    .statusCode(400);
+    }
+
+    @Test
+    @DisplayName("Return 400 given an invalid skillID")
+    public void testAPIReturn400_OnEndorsementSearchWithInvalidSkillID() {
```

Comment:
where is test for 403 and 500

Reply:
As per design document, we are not returning 403 in any scenario, how can I test this scenario?

Reply:
> As per design document, we are not returning 403 in any scenario, how can I test this scenario?

are you sure about this, pause and think before answering this..